scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -32,23 +32,43 @@
   ; suppress otherwise identical states so we don't spam.
   (= old-state new-state))
 
-(defn remove-finalization-if-set-and-delete
-  [api-client expected-state-dict pod]
-  (api/remove-finalization-if-set-and-delete api-client pod)
-  expected-state-dict)
+(defn delete-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail. Returns a new expected state
+  dict of nil."
+  [api-client pod]
+  (api/kill-task api-client pod)
+  nil) ; Return new expected state dict of nil.
 
 (defn kill-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail."
   [api-client expected-state-dict pod]
   (api/kill-task api-client pod)
   expected-state-dict)
 
+(defn weird
+  "Weird. This pod is in a weird state. Log its weird state."
+  [expected-state-dict pod]
+  (log/error "Pod in a weird state:" expected-state-dict "and pod" pod))
+
+(defn kill-task-weird
+  "We're in a weird state that shouldn't occur with any of the normal expected races. This 'shouldn't occur. However,
+  we're going to pessimistically assume that anything that could happen will, whether it shouldn't or not."
+  [api-client expected-state-dict pod]
+  (weird expected-state-dict pod)
+  (kill-task api-client expected-state-dict pod)
+  expected-state-dict)
+
 (defn launch-task
   [api-client expected-state-dict]
   (api/launch-task api-client expected-state-dict)
+  ; TODO: Should detect if we don't have a :launch-pod key and force a mea culpa retry and :expecte/killed so that we retry.
   expected-state-dict)
 
 (defn update-or-delete!
-  "Given a map atom, key, and value, if the value is not nil, set the key to the value in the map. If the value is nil, delete the key entirely."
+  "Given a map atom, key, and value, if the value is not nil, set the key to the value in the map. If the value is nil,
+  delete the key entirely."
```

Comment:
Nitpick: I prefer the new lines to be at sentence boundaries, so we could have inserted the new line before the `If`. This also makes it easier to move/insert sentences.